lib/multi-writable.js

```diff
@@ -59,43 +120,269 @@ class MultiWriteable extends stream.Writable {
+
+    if (indexOfValues === -1) {
+      this.next();
+    } else {
+      const tablePos = indexOfInsert + INSERT_INTO_BUFF.length;
+      const tableName = cleanVarName(this.buffer.slice(tablePos, indexOfValues).toString());
+      this.setTable(tableName);
+
+      this.bufferPosition = indexOfValues + VALUES_BUFF.length;
+      this.continueInsert();
+    }
+  }
+
+  continueInsert() {
     const writes = [];
-    let queueItem;
     let row;
-
-    for (let i = 0; i < nextWorkers.length; i += 1) {
-      queueItem = this.queue.shift();
-      row = this.processQueueItem(queueItem);
-      writes.push(this.writeToNextAvailableWorker(nextWorkers[i], row));
+    let worker;
+    let queueItem = this.getNextCommandParenSet();
+
+    while (
+      queueItem
+      && queueItem !== COMMAND_EXEC_BUFF
+      && this.readyWorkers.length
+    ) {
+      worker = this.readyWorkers.pop();
+      row = `[${queueItem}]`;
+      writes.push(this.writeToNextAvailableWorker(worker, row));
+
+      queueItem = this.getNextCommandParenSet();
     }
 
     Promise
       .all(writes)
       .then(() => {
-        if (this.queue.length) {
-          return this.flushQueueToAvailableWorkers(next);
+        if (!queueItem && this.bufferPosition >= this.buffer.length - 1) {
+          return this.next();
+        }
+
+        if (queueItem === COMMAND_EXEC_BUFF) {
+          this.cleanBuffer();
+          this.state = null;
+          return this.determineState();
         }
 
-        return next();
+        return this.continueInsert();
       })
       .catch((err) => {
         console.error(err);
       });
   }
 
+
+  // Handle Create Table State
+
+  runReadSchema() {
+    const indexOfCreate = this.buffer.indexOf(CREATE_TABLE_BUFF);
+    const indexOfNextParen = this.buffer.indexOf(BRACKET_OPEN_BUFF, indexOfCreate);
+    const startPos = indexOfCreate + CREATE_TABLE_BUFF.length;
+
+    if (indexOfNextParen === -1) {
+      this.next();
+    } else {
+      // set buffer head to the end of values
+      this.bufferPosition = startPos;
+
+      const tableName = cleanVarName(
+        this.buffer.slice(startPos + 1, indexOfNextParen).toString(),
+      );
+
+      const contents = this.getNextCommandParenSet();
+
+      if (contents) {
+        const createContents = contents.toString().trim();
+        const columns = buildSchema(createContents);
+        const schema = {
+          name: tableName,
+          columns,
+        };
+
+        this.tables[tableName] = schema;
+        this.sendSchema(schema);
+        console.log('Detected schema', JSON.stringify(schema, null, 2));
+
+
+        // Done processing Create Table command. Jump to end of line or as far as possible
+        const end = this.skipToEndOfCommand();
+
+        this.cleanBuffer();
+        this.state = null;
+
+        if (end === COMMAND_EXEC_BUFF) {
+          this.determineState();
+        } else if (!end) {
```

Comment:
Should perhaps a new state `SEEKING EOL` or something like that to find the end of command.